ignite/engine/engine.py

```diff
@@ -898,6 +901,8 @@ def switch_batch(engine):
                 # If engine was terminated and now is resuming from terminated state
                 # we need to initialize iter_counter as 0
                 self._init_iter = 0
+                # And we reset the skip_completed_after_termination to its default value
+                self.skip_completed_after_termination = False
```

Comment:
I wonder whether we really need to reset the flag here?

Reply:
I guess it would be useful in case one first terminates the training with `engine.terminate(True)`,  then for some reason resumes the training and wants to call `engine.terminate()` assuming `Events.COMPLETED` will be triggered. Do you think this is a realistic scenario?

Reply:
`self.skip_completed_after_termination` is always defined by the value passed in `engine.terminate(...)`, so if they resume and do not pass anything we'll internally overwrite `self.skip_completed_after_termination`. So, I do not see any problems in this case or missed something?

Reply:
Well, you are very right! Unless one manually fires `Events.TERMINATE` from an attached handler, I guess.